sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs

```diff
@@ -0,0 +1,211 @@
+
+            //var patch = new CommunicationServiceResourcePatch();
+            //patch.LinkedDomains.Add(_domainResource.Id.ToString());
+            //await _communicationService.UpdateAsync(patch);
+        }
+
+        [TearDown]
+        public async Task TearDown()
+        {
+            await foreach (var username in _domainResource.GetSenderUsernameResources().GetAllAsync())
+            {
+                if (!StringComparer.OrdinalIgnoreCase.Equals(username.Data.Name, "donotreply"))
+                {
+                    await username.DeleteAsync(WaitUntil.Completed);
+                }
+            }
+        }
+
+        [Test]
+        public async Task CreateOrUpdate()
+        {
+            var listName = "donotreply";
+            var suppressionList = await CreateDefaultSuppressionListResource(_domainResource, listName);
+
+            Assert.IsNotNull(suppressionList);
+            Assert.AreEqual(listName, suppressionList.Data.ListName);
+        }
+
+        [Test]
+        public async Task Delete()
+        {
+            var listName = "listToDelete";
+            var suppressionList = await CreateDefaultSuppressionListResource(_domainResource, listName);
+            var collection = _domainResource.GetSuppressionListResources();
+
+            var exists = await collection.ExistsAsync(suppressionList.Id.Name);
+            Assert.IsTrue(exists);
+
+            await suppressionList.DeleteAsync(WaitUntil.Completed);
+
+            collection = _domainResource.GetSuppressionListResources();
+            exists = await collection.ExistsAsync(suppressionList.Id.Name);
+            Assert.IsFalse(exists);
+        }
+
+        [Test]
+        public async Task Get()
+        {
+            var listName = "donotreply";
+            var suppressionList = await CreateDefaultSuppressionListResource(_domainResource, listName);
+            var collection = _domainResource.GetSuppressionListResources();
+            var actualSuppressionList = await collection.GetAsync(suppressionList.Data.Name);
+
+            Assert.IsNotNull(actualSuppressionList);
+            Assert.AreEqual(actualSuppressionList.Value.Data.ListName, listName);
+        }
+
+        [Test]
+        public async Task GetAll()
+        {
+            var listNames = new string[] { "list1", "list2", "list3" };
+            var resourceNames = new Dictionary<string, string>();
+
+            foreach (var listName in listNames)
+            {
+                var suppressionList = await CreateDefaultSuppressionListResource(_domainResource, listName);
+                resourceNames[suppressionList.Data.Name] = listName;
+            }
+
+            var suppressionLists = await _domainResource.GetSuppressionListResources().GetAllAsync().ToEnumerableAsync();
+
+            Assert.IsNotNull(suppressionLists);
+            Assert.IsTrue(suppressionLists.Count() >= listNames.Length);
```

Comment:
Why not check for exact count?

Reply:
Updated this scenario to use exact count.